File: C++/2540.cpp
#include <vector>
class Solution {
public:
    int getCommon(std::vector<int>& nums1, std::vector<int>& nums2) {
        int first{0}, second{0};

        while(first < nums1.size() && second < nums2.size()){
            if (nums1[first] == nums2[second]) return nums1[first];
            else if (nums1[first] > nums2[second]) second++;
            else first++;
        }

        return -1;
    }
};

// Topic: Two Pointers
// Time: O(n + m)
// Space: O(1)

File: Python/2540.py
class Solution:
    def getCommon(self, nums1: list[int], nums2: list[int]) -> int:
        first = 0
        second = 0

        while first < len(nums1) and second < len(nums2):
            if nums1[first] == nums2[second]:
                return nums1[first]
            elif nums1[first] > nums2[second]:
                second += 1
            else:
                first += 1
            
        return -1

# Topic: Two Pointers
# Time Complexity: O(n + m)
# Space Complexity: O(1)